src/components/BulkEnrollmentModal/index.jsx

```diff
@@ -53,24 +58,54 @@ class BulkEnrollmentModal extends React.Component {
     onClose();
   }
 
+  validateFormData(formData) {
+    const userEmailsKey = EMAIL_ADDRESS_TEXT_FORM_DATA;
+    const emailsCSVKey = EMAIL_ADDRESS_CSV_FORM_DATA;
+
+    /* eslint-disable no-underscore-dangle */
+    const errors = validateEmailAddressesFields(formData);
+    if (!formData[userEmailsKey] && !formData[emailsCSVKey]) {
+      const message = 'Either user emails or emails csv must be provided.';
+      errors[userEmailsKey] = message;
+      errors[emailsCSVKey] = message;
+      errors._error.push(message);
```

Comment:
So, I think doing validation this way is fighting the framework a bit. It looks like `redux-form` can take a `validate` function, and it would make sense to pull this validation code out of the main component (which is gr8 because that makes it easier to test, and then pass that validation function to redux-form: https://redux-form.com/8.3.0/examples/syncvalidation/

Reply:
when that website one day loads for me I will give this a go (or I'll give up and look into redux-form and the validate param/function elsewhere 😅 )